metricflow/model/validations/measures.py

```diff
@@ -364,6 +364,37 @@ def validate_model(model: UserConfiguredModel) -> List[ValidationIssueType]:  #
         return issues
 
 
+class CountAggregationExprRule(ModelValidationRule):
+    """Checks that COUNT measures have an expr provided."""
+
+    @staticmethod
+    @validate_safely(
+        whats_being_done="running model validation ensuring expr exist for measures with count aggregation"
+    )
+    def validate_model(model: UserConfiguredModel) -> List[ValidationIssueType]:  # noqa: D
+        issues: List[ValidationIssueType] = []
+
+        for data_source in model.data_sources:
+            for measure in data_source.measures:
+                if measure.agg == AggregationType.COUNT and measure.expr is None:
+                    issues.append(
+                        ValidationError(
+                            context=DataSourceElementContext(
+                                file_context=FileContext.from_metadata(metadata=data_source.metadata),
+                                data_source_element=DataSourceElementReference(
+                                    data_source_name=data_source.name, element_name=measure.name
+                                ),
+                                element_type=DataSourceElementType.MEASURE,
+                            ),
+                            message=(
+                                f"Measure '{measure.name}' is has a COUNT aggregation which requires an expr to be provided. "
```

Comment:
Do we need this in both places? @QMalcolm ?

Reply:
In both places meaning in the transformation and the validation? I think the answer is unfortunately yes. It shouldn't be that way, but currently the ordering of events is `Parsing -> Optionally Transforming -> Validating`. Because the transformation step is optional, we need the validation rule. But because if transforming happens, it happens prior to validation, it needs to be able to properly error out 😞

Reply:
Ok, I thought it might be something like this, thanks for confirming!